src/gui/accountstate.cpp

```diff
@@ -488,10 +488,14 @@ void AccountState::slotCredentialsAsked(AbstractCredentials *credentials)
 
     _waitingForNewCredentials = false;
 
-    if (!credentials->ready()) {
-        // User canceled the connection or did not give a password
+    switch (credentials->ready()) {
+    case AbstractCredentials::ReadyState::Retry:
+        return;
+    case AbstractCredentials::ReadyState::Error:
         setState(SignedOut);
         return;
+    default:
```

Comment:
Don't add default case. We wan't the compiler to warn when a new state is added.